src/view-backend.c

```diff
@@ -187,3 +197,31 @@ wpe_view_backend_dispatch_touch_event(struct wpe_view_backend* backend, struct w
     if (backend->input_client)
         backend->input_client->handle_touch_event(backend->input_client_data, event);
 }
+
+void
+wpe_view_backend_set_fullscreen_handler(struct wpe_view_backend* backend, void (*handler)(void*, bool), void* data)
```

Comment:
Same here: is it expected that this function would be called only once per view backend? Do we want to allow changing the handler?

We may want to cover the case in which fullscreening is requested by the application using WPE determines that it is not currently possible. I think the callback should have type `bool (*))(void*, bool)`; and if it returns `false`, cancel the fullscreening operation on the WebKit side.

Finally, it is unclear who has to implement a “client” and who has to provide a “handler”, having some documentation about what each does would bee great.

Reply:
I have improved the API with better function names, some comments and implementing the wait for the new exported image we talked about